src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -155,26 +122,40 @@ protected object ExecuteOperation(ResolveRequest request)
             {
                 InitiatingRequest = request;
 
-                _pipelineTracer?.OperationStart(this, request);
+                if (DiagnosticSource.OperationDiagnosticsEnabled())
```

Comment:
Suggest saving this as a boolean so we don't need to re-run the check at the end of the operation.

Reply:
Each diagnostic write now does `if(DiagnosticSource.IsEnabled() && DiagnosticSource.IsEnabled("eventname"))` before writing, which is basically a check to see if there are _any_ subscribers (a null check) prior to checking if a specific event is enabled. This should be roughly as fast as the null check we had before.